code/datums/station_traits/negative_traits.dm

```diff
@@ -134,3 +134,86 @@
 		/// The bot's language holder - so we can randomize and change their language
 		var/datum/language_holder/bot_languages = found_bot.get_language_holder()
 		bot_languages.selected_language = bot_languages.get_random_spoken_language()
+
+/datum/station_trait/revenge_of_pun_pun
+	name = "Revenge of Pun Pun"
+	trait_type = STATION_TRAIT_NEGATIVE
+	weight = 2
+	var/static/list/weapon_types
+
+/datum/station_trait/revenge_of_pun_pun/New()
+	if(!weapon_types)
+		weapon_types = list(
+			/obj/item/chair = 20,
+			/obj/item/tailclub = 10,
+			/obj/item/melee/baseball_bat = 10,
+			/obj/item/melee/chainofcommand/tailwhip = 10,
+			/obj/item/melee/chainofcommand/tailwhip/kitty = 10,
+			/obj/item/reagent_containers/food/drinks/bottle = 20,
+			/obj/item/reagent_containers/food/drinks/bottle/kong = 5,
+			/obj/item/switchblade/extended = 10,
+			/obj/item/sign/random = 10,
+			/obj/item/gun/ballistic/automatic/pistol = 1,
+		)
+
+	RegisterSignal(SSatoms, COMSIG_SUBSYSTEM_POST_INITIALIZE, .proc/arm_monke)
+
+/datum/station_trait/revenge_of_pun_pun/proc/arm_monke()
+	SIGNAL_HANDLER
+	var/mob/living/carbon/human/species/monkey/punpun/punpun = locate()
+	if(!punpun)
+		return
+	var/weapon_type = pickweight(weapon_types)
+	var/obj/item/weapon = new weapon_type
+	if(!punpun.put_in_l_hand(weapon) && !punpun.put_in_r_hand(weapon))
+		// Guess they did all this with whatever they have in their hands already
+		qdel(weapon)
+
+	weapon?.add_mob_blood(punpun)
+	punpun.add_mob_blood(punpun)
+
+	punpun.ai_controller = /datum/ai_controller/monkey/angry
```

Comment:
Is Punpun already initialised at this point? I'm going to assume yes if you're already calling its procs.

You may want to just do `new /datum/ai_controller/monkey/angry(punpun)` as otherwise you'll be overriding Punpun's instantiated AI controller datum (which was new'd in `atom.Initialize()` using `InitializeAIController()`) with a type path, possibly causing unintended side effects with pawn possession etc. by the AI controllers.

Reply:
Yes pun pun is already initialized as this is set to trigger after SSatoms is done initializing but this should be changed to be handled properly.